ext/atomic/index.md

```diff
@@ -0,0 +1,510 @@
+`204 No Content` and no response document if no operations are required to
+return `data`.
+
+### <a href="#processing-errors" id="processing-errors" class="headerlink"></a> Processing Errors
+
+If any operation in a request fails, the server **MUST** respond as described
+[in the base specification](/format/#errors-processing). An array of one or
+more [error objects](/format/#error-objects) **SHOULD** be returned, each with
+a `source` member that contains a `pointer` to the source of the problem in the
+request document.
+
+### <a href="#specific-operations" id="specific-operations" class="headerlink"></a> Processing Specific Operations
+
+The following sections describe how to process specific operations.
+
+#### Creating Resources
+
+An operation that creates a resource **MAY** target a resource collection
+through the operation's `ref` member. The operation **MUST** include an `op`
+code of `"add"` as well as a resource object as `data`. The resource object
+**MUST** contain at least a `type` member.
+
+For example:
+
+```json
+POST /operations HTTP/1.1
+Host: example.org
+Content-Type: application/vnd.api+json;ext="https://jsonapi.org/ext/atomic"
+Accept: application/vnd.api+json;ext="https://jsonapi.org/ext/atomic"
+
+{
+  "atomic:operations": [{
+    "op": "add",
+    "href": "/blogPosts",
+    "data": {
+      "type": "articles",
+      "attributes": {
+        "title": "JSON API paints my bikeshed!"
+      }
+    }
+  }]
+}
+```
+
+Note that `href` is used in this example to target a resource collection,
+`blogPosts` in this case, that is distinct from the `type` of the resource.
+
+##### Responses
+
+If the server is able to create a resource with a client-generated ID and its
+representation would be identical to the resource in the operation, then the
+server **MAY** return a result with no `data` member or it **MAY** return a
+result that contains the created resource as `data`.
+
+In all other cases in which the server is able to successfully create the
+resource, the server **MUST** return a result that contains the created resource
+as `data`.
+
+For example:
+
+```json
+HTTP/1.1 200 OK
+Content-Type: application/vnd.api+json;ext="https://jsonapi.org/ext/atomic"
+
+{
+  "atomic:results": [{
+    "data": {
+      "links": {
+        "self": "http://example.com/blogPosts/13"
+      },
+      "type": "articles",
+      "id": "13",
+      "attributes": {
+        "title": "JSON API paints my bikeshed!"
+      }
+    }
+  }]
+}
+```
+
+If the server is unable to create the resource, an appropriate error response
+**MUST** be returned and a response document **SHOULD** be returned that
+contains a top-level `errors` as [described above](#processing-errors).
+
+#### Updating Resources
+
+An operation that updates a resource **MUST** target that resource through the
+operation's `ref` member. The operation **MUST** include an `op` code of
+`"update"`.
+
+For example:
+
+```json
+POST /bulk HTTP/1.1
```

Comment:
@dgeb /bulk occurs multiple times; you may want to update remaining references to /operations too.

Reply:
@bart-degreed good call - I've just updated them all.